schema/encoding/jsonschema/all_test.go

```diff
@@ -0,0 +1,216 @@
+					},
+					"age": {
+						"type": "integer"
+					}
+				},
+				"required": ["name"]
+			}`,
+		},
+		{
+			name: "Required=true,Required=true",
+			schema: schema.Schema{
+				Fields: schema.Fields{
+					"name": schema.Field{
+						Required:  true,
+						Validator: &schema.String{},
+					},
+					"age": schema.Field{
+						Required:  true,
+						Validator: &schema.Integer{},
+					},
+				},
+			},
+			customValidate: func(t *testing.T, result []byte) {
+				var expectProperties = map[string]interface{}{
+					"name": map[string]interface{}{
+						"type": "string",
+					},
+					"age": map[string]interface{}{
+						"type": "integer",
+					},
+				}
+
+				v := make(map[string]interface{})
+				err := json.Unmarshal(result, &v)
+				assert.NoError(t, err, "Input ('%s') needs to be valid JSON", result)
+
+				assert.Equal(t, "object", v["type"], `Unexpected "type" value`)
+				assert.Equal(t, expectProperties, v["properties"], `Unexpected "properties" value`)
+				assert.Len(t, v["required"], 2, `Unexpected "required" value`)
+				assert.IsType(t, []interface{}{}, v["required"], `Unexpected "required" value`)
+				assert.Contains(t, v["required"], "name", `Unexpected "required" value`)
+				assert.Contains(t, v["required"], "age", `Unexpected "required" value`)
+			},
+		},
+		// Documenting the current behavior, but unsure what's the best approach. schema.Object with no schema
+		// appears to be invalid and will cause an error if used.
+		{
+			name: "type=Array,ValuesValidator=Object{Schema:nil}",
+			schema: schema.Schema{
+				Fields: schema.Fields{
+					"students": schema.Field{
+						Validator: &schema.Array{
+							ValuesValidator: &schema.Object{},
+						},
+					},
+				},
+			},
+			expect: `{
+				"type": "object",
+				"properties": {
+					"students": {
+						"type": "array",
+						"items": {}
+					}
+				}
+			}`,
+		},
+		{
+			name: "type=Array,ValuesValidator=Object{Schema:Student}",
+			schema: schema.Schema{
+				Description: "A list of students",
+				Fields: schema.Fields{
+					"students": schema.Field{
+						Validator: &schema.Array{
+							ValuesValidator: &schema.Object{
+								Schema: &schema.Schema{
+									Fields: schema.Fields{
+										"student": schema.Field{
+											Description: "a student",
+											Required:    true,
+											Default:     "Unknown",
+											Validator: &schema.String{
+												MinLen: 1,
+												MaxLen: 10,
+											},
+										},
+										"class": schema.Field{
+											Default: "Unassigned",
+											Validator: &schema.String{
+												MinLen: 0, // Default value.
+												MaxLen: 10,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			// FIXME: "object" is the only field-type that encodes description as "title" instead of
+			// "description". Should it be encoded as "description" here as well?
+			expect: `{
+				"type": "object",
+				"title": "A list of students",
```

Comment:
I am wondering if we should switch this to "description". @yanfali, do you have an opinion? Is it any reason in particular that you chose to encode the schema description as "title" for "object", while it's encoded as "description" for other types?

JSON Schema defines both "title" and "description", where title is supposed to be shorter, and description is supposed to be longer.